### PR TITLE
refactor: Remove wrapper test components in button tests (#2876)

### LINKED ISSUE #2876: Redundant Test components for button components in the Bit BlazorUI Test project
For each button component in the Test project, a Test component is added which currently there is no use for them. in order to have a simpler testing experience, it is better to remove them, and use the component directly in the unit test methods.

### PR DESCRIPTION
**Pull request by linkdotnet**
this closes #2876 

This is a follow up PR to #2872.

I took the liberty to remove all the wrapper test components in button-tests and inject the cascading parameter directly. Making the test easier to understand and maintain. 

Let me know your thoughts.

### CODE REVIEW

**Review by msynk — APPROVED** [review]

**Comment by msynk** [comment]
@Cyrus-Sushiant plz review

@linkdotnet this looks good, tnx 👍
I think we should go with this approach for other components too if it's possible. since I know there are some tests that require this Test component.

File: src/BlazorUI/Bit.BlazorUI.Tests/Buttons/BitIconButtonTests.cs [hunk 1]
```diff
@@ -5,7 +5,7 @@
 namespace Bit.BlazorUI.Tests.Buttons;
 
 [TestClass]
-public class BitIconButtonTests : BunitTestContext
+public class BitIconButtons : BunitTestContext
```
Comment (Cyrus-Sushiant) [inline src/BlazorUI/Bit.BlazorUI.Tests/Buttons/BitIconButtonTests.cs:8]: This is the same as before, you should put the "Tests" at the end of the class name.

File: src/BlazorUI/Bit.BlazorUI.Tests/Buttons/BitToggleButtonTests.cs [hunk 2]
```diff
@@ -4,7 +4,7 @@
 namespace Bit.BlazorUI.Tests.Buttons;
 
 [TestClass]
-public class BitToggleButtonTests : BunitTestContext
+public class BitToggleButtons : BunitTestContext
```
Comment (Cyrus-Sushiant) [inline src/BlazorUI/Bit.BlazorUI.Tests/Buttons/BitToggleButtonTests.cs:7]: This is the same as before, you should put the "Tests" at the end of the class name.

**Review by Cyrus-Sushiant — CHANGES_REQUESTED** [review]

File: src/BlazorUI/Bit.BlazorUI.Tests/Buttons/BitCompoundButtonTests.cs [hunk 3]
```diff
@@ -4,7 +4,7 @@
 namespace Bit.BlazorUI.Tests.Buttons;
 
 [TestClass]
-public class BitCompoundButtonTests : BunitTestContext
+public class BitCompoundButtons : BunitTestContext
```
[the remark this answers is not in the document]
Comment (msynk) [reply src/BlazorUI/Bit.BlazorUI.Tests/Buttons/BitCompoundButtonTests.cs:7]: I agree 👍

File: src/BlazorUI/Bit.BlazorUI.Tests/Buttons/BitToggleButtonTests.cs (hunk 2 above)
Comment (linkdotnet) [reply src/BlazorUI/Bit.BlazorUI.Tests/Buttons/BitToggleButtonTests.cs]: That was on me.. done.

File: src/BlazorUI/Bit.BlazorUI.Tests/Buttons/BitIconButtonTests.cs (hunk 1 above)
Comment (linkdotnet) [reply src/BlazorUI/Bit.BlazorUI.Tests/Buttons/BitIconButtonTests.cs]: done

File: src/BlazorUI/Bit.BlazorUI.Tests/Buttons/BitCompoundButtonTests.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (linkdotnet) [reply src/BlazorUI/Bit.BlazorUI.Tests/Buttons/BitCompoundButtonTests.cs]: Yeah search and replace a bit too much ... done ;)

**Review by msynk — APPROVED** [review]

**Pull request merged**